### ISSUE TITLE
Database tests should be self contained

### ISSUE THREAD

**Issue by mirrormind** [mirrormind/mo-test-eam#6]
At the moment, running a test against the db query classes requires the tester to manually run (in the correct order) the SQL scripts provided, set up some data, then run the appropriate tests. This will become an inconvenience when running tests for topics and posts since they will require existing data from the "upper hierarchy" of the forum (created at the expense of the tester's time).

The tests should be rewritten so that it can set up the required tables without interfering with existing tables, create its own data and perform any setup for the test to run, and finally clean up after itself by deleting any table or entries that may or may not be created for the purpose of the test.

In summary:
* Modify the SQL scripts (the actual .sql files) so that all tables can be created in one go
* Change the tests so that it can create its own tables and data prior to actually running tests
* Clean up after itself when it's done
* Consider creating a test suite

**Comment by mirrormind** [comment]
https://github.com/mirrormind/mo-test-eam/commit/1e10c80

Just realised we'll need to pull out the table names (and most likely all the table column names) before self contained tests will be a thing. At the moment the scripts assign a hardcoded (facepalm) table name so we won't be able to dynamically create tables AND use them with our current code path. I have an idea for an intelligent SQL script builder (which as a side effect will solve the issue of self-contained tests), but it's a decent project on its own.

Fyi, I really want someone to work on this framework because I don't want to be manually creating tables, deleting entries, etc every time I want to know if a new piece of code is working or not. This whole testing process from start to finish should be AUTOMATED.
